src/auth/models.hpp

```diff
@@ -36,17 +36,18 @@ enum class Permission : uint64_t {
   TRIGGER     = 1U << 14U,
   CONFIG      = 1U << 15U,
   AUTH        = 1U << 16U,
-  STREAM      = 1U << 17U
+  STREAM      = 1U << 17U,
+  WEBSOCKET   = 1U << 18U
 };
 // clang-format on
 
 // Constant list of all available permissions.
-const std::vector<Permission> kPermissionsAll = {Permission::MATCH,      Permission::CREATE,    Permission::MERGE,
-                                                 Permission::DELETE,     Permission::SET,       Permission::REMOVE,
-                                                 Permission::INDEX,      Permission::STATS,     Permission::CONSTRAINT,
-                                                 Permission::DUMP,       Permission::AUTH,      Permission::REPLICATION,
-                                                 Permission::DURABILITY, Permission::READ_FILE, Permission::FREE_MEMORY,
-                                                 Permission::TRIGGER,    Permission::CONFIG,    Permission::STREAM};
+const std::vector<Permission> kPermissionsAll = {
```

Comment:
This has `internal_linkage` by default because of `const` which means that a vector is created on each translation unit. I know this is a little beyond the scope of this PR but would you mind adding a more idiomatic way like:
```
inline const auto = []() -> const std::array<Permision, 19>& { 
  static std::array<Permission, 19> cache;
  static bool once = true;
  if(once) [[unlikely]] {
    std::generate(cache.begin(), cache.end(), [n = 0] { return static_cast<Permission>(n++); //watch post increment });
    once = false;
  }
  return cache_;
};
```

p.s. with reflection we could extract the total number of fields in the enum and abstract away the size of the array (not hardcoded with 19) which would then introduce a factory method for every use of the above pattern.

Reply:
This approach requires that I need to call `kPermissionsAll()` in order to get the ref to it, it's kind of weird and complicated to me. Can't we just make it `cosntexpr std::array`?

Reply:
Maybe using `constexpr std::array` instead? Or as this vector is only used in the source file, move it to the source file.